Modix.Data.Test/Repositories/EmojiRepositoryTests.cs

```diff
@@ -0,0 +1,300 @@
+            emoji.UserId.ShouldBe(data.UserId);
+            emoji.EmojiId.ShouldBe(data.EmojiId);
+            emoji.EmojiName.ShouldBe(data.EmojiName);
+            emoji.IsAnimated.ShouldBe(data.IsAnimated);
+            emoji.Timestamp.ShouldBeInRange(
+                DateTimeOffset.Now - TimeSpan.FromSeconds(1),
+                DateTimeOffset.Now + TimeSpan.FromSeconds(1));
+            emoji.UsageType.ShouldBe(data.UsageType);
+
+            modixContext.Emoji.Where(x => x.Id != emoji.Id).Select(x => x.Id).ShouldBe(Emoji.Entities.Select(x => x.Id));
+            modixContext.Emoji.Where(x => x.Id != emoji.Id).EachShould(x => x.ShouldNotHaveChanged());
+
+            await modixContext.ShouldHaveReceived(1)
+                .SaveChangesAsync();
+        }
+
+        #endregion CreateAsync() Tests
+
+        #region CreateMultipleAsync() Tests
+
+        [Test]
+        public async Task CreateMultipleAsync_DataIsNull_DoesNotUpdateEmojisAndThrowsException()
+        {
+            (var modixContext, var uut) = BuildTestContext();
+
+            await Should.ThrowAsync<ArgumentNullException>(uut.CreateMultipleAsync(null, 1));
+
+            modixContext.Emoji.Select(x => x.Id).ShouldBe(Emoji.Entities.Select(x => x.Id));
+            modixContext.Emoji.EachShould(x => x.ShouldNotHaveChanged());
+
+            await modixContext.ShouldNotHaveReceived()
+                .SaveChangesAsync();
+        }
+
+        [TestCaseSource(nameof(EmojiCreationTestCases))]
+        public async Task CreateMultipleAsync_CountIsZero_DoesNotUpdateEmojis(EmojiCreationData data)
+        {
+            (var modixContext, var uut) = BuildTestContext();
+
+            await uut.CreateMultipleAsync(data, 0);
+
+            modixContext.Emoji.Select(x => x.Id).ShouldBe(Emoji.Entities.Select(x => x.Id));
+            modixContext.Emoji.EachShould(x => x.ShouldNotHaveChanged());
+
+            await modixContext.ShouldNotHaveReceived()
+                .SaveChangesAsync();
+        }
+
+        [TestCaseSource(nameof(EmojiCreationTestCases))]
+        public async Task CreateMultipleAsync_CountIsNegative_DoesNotUpdateEmojis(EmojiCreationData data)
+        {
+            (var modixContext, var uut) = BuildTestContext();
+
+            await uut.CreateMultipleAsync(data, -1);
+
+            modixContext.Emoji.Select(x => x.Id).ShouldBe(Emoji.Entities.Select(x => x.Id));
+            modixContext.Emoji.EachShould(x => x.ShouldNotHaveChanged());
+
+            await modixContext.ShouldNotHaveReceived()
+                .SaveChangesAsync();
+        }
+
+        [TestCaseSource(nameof(EmojiCreationTestCases))]
+        public async Task CreateMultipleAsync_DataIsNotNull_InsertsEmojis(EmojiCreationData data)
+        {
+            const int RecordCount = 100;
+
+            (var modixContext, var uut) = BuildTestContext();
+
+            await uut.CreateMultipleAsync(data, RecordCount);
+
+            var emojis = modixContext.Emoji
```

Comment:
Could we instead have the method return the ID's it generated, like the `CreateAsync()` method does?